src/mesh/python/x3d_generator.py

```diff
@@ -178,7 +184,8 @@
 # -- assume two boundaries per dimension
 # -- x3d only needs a unique node list
 for i in range(2 * mesh.ndim):
-    np.savetxt(fname + '.bdy' + str(i + 1) + '.in', np.unique(mesh.nodes_per_side[i] + 1), fmt='%d')
+    np.savetxt(fname + '.bdy' + str(i + 1) + '.in',
+               np.unique(np.array(mesh.nodes_per_side[i]) + 1), fmt='%d')
```

Comment:
I *think* these are benign changes to `x3d_generator.py` but if you disagree let's iterate @RyanWollaeger -- I did have to change a few things around since for example with varying numbers of nodes for boundary I couldn't just express `mesh.nodes_per_side` as a 2D numpy `array`.

Reply:
Ohhh - that was a bug in the previous version then, since variable side (boundary) nodes should've definitely been possible (I must've missed testing it - sorry!).

Reply:
> What was the issue that precluded differing number of nodes per boundary?

This was that issue